pkg/sharings/send_mails.go

```diff
@@ -76,15 +72,30 @@ func SendSharingMails(instance *instance.Instance, s *Sharing) error {
 		// are of no use in this situation.
 		// FI: they correspond to the job information and to a channel with
 		// which we can check the advancement of said job.
-		_, _, err = instance.JobsBroker().PushJob(&jobs.JobRequest{
+		_, _, errJobs := instance.JobsBroker().PushJob(&jobs.JobRequest{
 			WorkerType: "sendmail",
 			Options:    nil,
 			Message:    sharingMessage,
 		})
-		if err != nil {
-			errorOccurred = logErrorAndSetRecipientStatus(rs, err)
+		if errJobs != nil {
+			errorOccurred = logError(errJobs)
 			continue
 		}
+
+		// Job was created, we set the status to "pending".
+		rs.Status = consts.PendingSharingStatus
+	}
+
+	// Persist the modifications in the database.
+	err = couchdb.UpdateDoc(instance, s)
+	if err != nil {
+		if errorOccurred {
```

Comment:
The idea here is to return an error if one of the email could not be sent, and that error must say so. Now that I think about it this is probably not the best way to do it as it doesn't give much information (doesn't say for whom), and as one error was already logged before.